addons/survey/static/src/js/survey_result.js

```diff
@@ -497,10 +518,70 @@ publicWidget.registry.SurveyResultWidget = publicWidget.Widget.extend({
      * @param {Event} ev
      */
     _onFilterFinishedClick: function (ev) {
-        var params = new URLSearchParams(window.location.search);
-        params.set('finished', true);
+        let params = new URLSearchParams(window.location.search);
+        params.set('finished', 'true');
+        window.location.href = window.location.pathname + '?' + params.toString();
+    },
+
+    /**
+     * @private
+     * @param {Event} ev
+     */
+    _onFilterFailedClick: function (ev) {
+        let params = new URLSearchParams(window.location.search);
+        params.set('failed', 'true');
+        params.delete('passed');
+        window.location.href = window.location.pathname + '?' + params.toString();
+    },
+
+    /**
+     * @private
+     * @param {Event} ev
+     */
+    _onFilterPassedClick: function (ev) {
+        let params = new URLSearchParams(window.location.search);
+        params.set('passed', 'true');
+        params.delete('failed');
         window.location.href = window.location.pathname + '?' + params.toString();
     },
+
+    /**
+     * @private
+     * @param {Event} ev
+     */
+    _onFilterPassedAndFailedClick: function (ev) {
+        let params = new URLSearchParams(window.location.search);
+        params.delete('failed');
+        params.delete('passed');
+        window.location.href = window.location.pathname + '?' + params.toString();
+    },
+
+    /**
+     * Returns the modified pathname string for filters after adding or removing an
+     * answer filter (from click event). Filters are formatted as `"rowX,ansX", where
+     * the row is used for matrix-type questions and set to 0 otherwise.
+     * @private
+     * @param {String} filters Existing answer filters, formatted as `rowX,ansX|rowY,ansY...`.
+     * @param {("add", "remove")} operation Whether to add or remove the filter.
```

Comment:
Not sure `@param {("add", "remove")}` is a valid type.

As the documentation is generated using a tool (don't remember the name), we need to keep types readable, maybe replace by "String" and mention the possible values in the sentence following the type?

Reply:
My IDE understands this. I'll look into what Odoo doc uses.
![operation](https://user-images.githubusercontent.com/90857153/146539921-6e300e89-412c-4748-9df1-5a900ce7263e.png)

Reply:
I believe it's "sphinx" but not 100% sure, I have not used it recently.

Reply:
I couldn't find a reference to the tool to test it. However, I found 10 other occurrences of this in the codebase (without the brackets in this case, so I removed them).
IDEs now recognize this too (I especially checked VS Code too).

![operation_vscode](https://user-images.githubusercontent.com/90857153/146560452-55f0af3f-d400-4305-b5ae-ab43a0c22044.png) 

JSDoc and Sphinx-js support multiple types and these will be resolved into `String` as they are unambiguous. But I'd understand if you'd prefer 
``` js
/*
* @param {String} operation Whether to add or remove the filter ("add" or "remove").
*/
```

Thumb up or down to keep pseudo-enum :+1: :-1:  ?

Reply:
I let you decide, if you are ready to face the couroux of the js team if the doc building doesn't work :smile:

Reply:
I should have an answer shortly from my contact in the team ;-)